util/uuid.c

```diff
@@ -1,34 +1,45 @@
-#include <uuid.h>
+#define _DEFAULT_SOURCE
 #include "util/uuid.h"
 
-#if HAS_LIBUUID
-bool generate_uuid(char out[static 37]) {
-	uuid_t uuid;
-	uuid_generate_random(uuid);
-	uuid_unparse(uuid, out);
-	return true;
-}
-#else
-#include <assert.h>
+#include <inttypes.h>
+#include <stdint.h>
+#include <stdio.h>
 #include <string.h>
-#include <stdlib.h>
+#include <unistd.h>
+
+#ifndef HAS_GETENTROPY
+static int getentropy(void *buffer, size_t len)
+{
+	FILE *urandom;
+	int ret = -1;
+
+	if ((urandom = fopen("/dev/urandom", "r"))) {
+		if (fread(buffer, 1, len, urandom) != 1) {
+			ret = 0;
+		}
+		fclose(urandom);
+	}
+
+	return ret;
+}
+#endif
+
+#define UUID_FORMAT "%04" PRIx16 "%04" PRIx16 \
+	"-%04" PRIx16 \
+	"-%04" PRIx16 \
+	"-%04" PRIx16 \
+	"-%04" PRIx16 "%04" PRIx16 "%04" PRIx16
+bool generate_uuid(char out[static 37])
+{
+	uint16_t data[8];
 
-bool generate_uuid(char out[static 37]) {
-	uuid_t uuid;
-	uint32_t status;
-	uuid_create(&uuid, &status);
-	if (status != uuid_s_ok) {
+	if (getentropy(data, 16)) {
```

Comment:
`sizeof(data)` instead of hardcoding 16, so that it's easy to know where this number comes from.